File: src/main/scala/demo/sortDemo/demo1.scala
package demo.sortDemo

import org.apache.spark.{SparkConf, SparkContext}

/**
  * Created by wanglei on 2018/1/1.
  * 案例需求：
  * 1、对文本文件内的每个单词都统计出其出现的次数
  * 2、按照每个单词出现次数的数量，降序排序
  */
object demo1 {
  def main(args: Array[String]): Unit = {
    val sparkConf = new SparkConf().setAppName(this.getClass.getName.replace("$", "")).setMaster("local[1]")
    val sparkContext = new SparkContext(sparkConf)
    val rdd = sparkContext.textFile("F:\\IdeaProjects\\MySpark\\src\\testData\\hello.txt")

    //统计单词出现的次数
    val words = rdd.flatMap(line => line.split(" "))
    val tupleRDD = words.map(word => (word, 1))
    val result = tupleRDD.reduceByKey(_ + _)
    result.foreach(println(_))
    println("=======================================")

    //按键排序
    val revTuple = result.map(tuple => (tuple._2, tuple._1))
    val resultSorted = revTuple.sortByKey()
    resultSorted.foreach(println(_))
  }
}

File: src/main/java/PlayCode/Demo/打印/Demo1.java
package PlayCode.Demo.打印;
//1+2+3+4+5+6+7+8+9+10=55
//21+22+23+24+25+26+27+28+29+30=255
//....
//90+91+..............+100=955
//*/

class Demo1 {
	public static void main(String[] args) {
		int res=0;
		for(int i=1;i<=100;i++){
			if(i%10!=0){
				res+=i;
				System.out.print(i+"+");
			}else{
				res+=i;
				System.out.println(i+"="+res);
				res=0;
			}
		}
	}
}


